service/src/main/java/bio/terra/landingzone/library/landingzones/management/ResourcesReaderImpl.java

```diff
@@ -59,13 +53,43 @@ public List<DeployedResource> listResourcesByPurpose(ResourcePurpose purpose) {
         purpose.toString());
   }
 
+  @Override
+  public List<DeployedResource> listResourcesWithPurpose() {
+    List<ResourcePurpose> supportedPurposes = ResourcePurpose.values().stream().toList();
+    return listResourceByTag(
+            resourceGroup.name(), LandingZoneTagKeys.LANDING_ZONE_PURPOSE.toString(), null)
+        .stream()
+        .filter(
+            deployedResource ->
+                supportedPurposes.contains(
+                    ResourcePurpose.fromString(
+                        deployedResource
+                            .tags()
+                            .get(LandingZoneTagKeys.LANDING_ZONE_PURPOSE.toString()))))
+        .toList();
+  }
+
   @Override
   public List<DeployedVNet> listVNetWithSubnetPurpose(SubnetResourcePurpose purpose) {
     return listResourceByTag(resourceGroup.name(), purpose.toString(), null).stream()
         .map(this::toDeployedVNet)
         .collect(Collectors.toList());
   }
 
+  @Override
+  public List<DeployedVNet> listVNetResourcesWithSubnetPurpose() {
```

Comment:
Similar to `listResourcesWithPurpose`, do we want to check the `LANDING_ZONE_PURPOSE` tag is a valid value here?

Reply:
Network tag for VNet has a different structure: purpose to a subnet name. No LANDING_ZONE_PURPOSE involved.